File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/comment/comment/direct/Java/Python/s527301188_commented.py
import sys

def main():
    import sys
    input = sys.stdin.read().strip()
    strs = input.split()
    nums = []
    for s in strs:
        if s == '+':
            b = nums.pop()
            a = nums.pop()
            nums.append(a + b)
        elif s == '-':
            b = nums.pop()
            a = nums.pop()
            nums.append(a - b)
        elif s == '*':
            b = nums.pop()
            a = nums.pop()
            nums.append(a * b)
        else:
            nums.append(int(s))
    print(nums[0])

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s527301188_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, err := reader.ReadString('\n')
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error reading input:", err)
		os.Exit(1)
	}
	strs := strings.Split(strings.TrimSpace(input), " ")
	nums := make([]int, len(strs))
	top := 0

	for i := 0; i < len(strs); i++ {
		switch strs[i][0] {
		case '+':
			top--
			nums[top-1] = nums[top-1] + nums[top]
		case '-':
			top--
			nums[top-1] = nums[top-1] - nums[top]
		case '*':
			top--
			nums[top-1] = nums[top-1] * nums[top]
		default:
			num, err := strconv.Atoi(strs[i])
			if err != nil {
				fmt.Fprintln(os.Stderr, "Error parsing number:", err)
				os.Exit(1)
			}
			nums[top] = num
			top++
		}
		//fmt.Println(nums)
	}
	fmt.Println(nums[0])
}

// End of Code
